pkg/generate/app/app_test.go

```diff
@@ -257,3 +258,244 @@ func TestImageStream(t *testing.T) {
 		}
 	}
 }
+
+func TestNameSuggestions_SuggestName(t *testing.T) {
+	tests := map[string]struct {
+		nameSuggestions NameSuggestions
+		expectedName    string
+		expectedSuccess bool
+	}{
+		"good suggestion from first": {
+			nameSuggestions: []NameSuggester{&suggestWith{"foo", true}, &suggestWith{"", false}},
+			expectedName:    "foo",
+			expectedSuccess: true,
+		},
+		"good suggestion from second": {
+			nameSuggestions: []NameSuggester{&suggestWith{"foo", false}, &suggestWith{"bar", true}},
+			expectedName:    "bar",
+			expectedSuccess: true,
+		},
+		"no good suggestions": {
+			nameSuggestions: []NameSuggester{&suggestWith{"foo", false}, &suggestWith{"bar", false}},
+			expectedName:    "",
+			expectedSuccess: false,
+		},
+		"nil suggestion": {
+			nameSuggestions: []NameSuggester{nil, &suggestWith{"bar", true}},
+			expectedName:    "bar",
+			expectedSuccess: true,
+		},
+	}
+
+	for name, test := range tests {
+		suggestedName, success := test.nameSuggestions.SuggestName()
+		if suggestedName != test.expectedName {
+			t.Errorf("%s expected name %s but recieved %s", name, test.expectedName, suggestedName)
+		}
+		if success != test.expectedSuccess {
+			t.Errorf("%s expected success condition %t but recieved %t", name, test.expectedSuccess, success)
+		}
+	}
+}
+
+type suggestWith struct {
+	name    string
+	success bool
+}
+
+func (s *suggestWith) SuggestName() (string, bool) {
+	return s.name, s.success
+}
+
+func TestIsParameterizableValue(t *testing.T) {
+	tests := []struct {
+		val            string
+		expectedReturn bool
+	}{
+		{"foo", false},
+		{"{foo}", false},
+		{"$foo}", false},
+		{"foo}", false},
+		{"{foo", false},
+		{"${foo", true},
```

Comment:
Sigh... this makes me 😢, why we allow this? Is it for supporting spaces inside of those or what? @bparees